### PR TITLE
[SPARK-35541][SQL] Simplify OptimizeSkewedJoin

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Various small code simplification/cleanup for OptimizeSkewedJoin

### Why are the changes needed?

code refactor

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala [hunk 1]
```diff
@@ -174,20 +175,20 @@ object OptimizeSkewedJoin extends CustomShuffleReaderRule {
...
+    val rightTargetSize = targetSize(rightSizes, rightSkewThreshold)
 
     val leftSidePartitions = mutable.ArrayBuffer.empty[ShufflePartitionSpec]
     val rightSidePartitions = mutable.ArrayBuffer.empty[ShufflePartitionSpec]
     var numSkewedLeft = 0
     var numSkewedRight = 0
     for (partitionIndex <- 0 until numPartitions) {
       val leftSize = leftSizes(partitionIndex)
-      val isLeftSkew = isSkewed(leftSize, leftMedSize) && canSplitLeft
+      val isLeftSkew = canSplitLeft && leftSize > leftSkewThreshold
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala:189]: now we don't need to call `isSkewed` repeatedly

**Comment by cloud-fan** [comment]
cc @yaooqinn @viirya

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala [hunk 2]
```diff
@@ -251,8 +251,7 @@ object OptimizeSkewedJoin extends CustomShuffleReaderRule {
 
     case shj @ ShuffledHashJoinExec(_, _, joinType, _, _,
         ShuffleStage(left: ShuffleQueryStageExec),
-        ShuffleStage(right: ShuffleQueryStageExec), isSkewJoin)
-        if !isSkewJoin && supportedJoinTypes.contains(joinType) =>
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala:255]: We don't need to check the join types now, as `tryOptimizeJoinChildren` will return None immediately if the join type is not supported.

**Review by yaooqinn — APPROVED** [review]
LGTM.

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Thanks! Merging to master!

**Pull request closed without merging**